### ISSUE TITLE
[MU4 Issue] No way to mute individual voices

### ISSUE THREAD

**Issue by SamYeh** [musescore/MuseScore#14677]
**Describe the bug**
MU4 lacks the functionality to mute individual voices in a stave, which MU3 supports.

**Expected behavior**
User should be able to mute individual voices like in MU3, shown below:
![image](https://user-images.githubusercontent.com/10248807/203446504-ecf67f62-4aa1-4f3f-b296-c7a9797b3a6f.png)

Note: Am I missing something? I searched in GitHub, the musescore.org forum, and MU4 handbook and could not find anything related. Is this something that is planned for MU4 release / post-release?

**Comment by Tantacrul** [comment]
This was a MIDI-based capability we had to drop for 4.0 but will re-introduce in later versions.

For context, MS3 was a MIDI-based app. MS4 does not use MIDI for its primary playback anymore. When using the Sound Font, this may seem like it's just the same thing with a few minor differences - but it was necessary for a variety of things we want to achieve in the future (as was something that allows Muse Sounds to sound so good too).

With this in mind, there were some things we just didn't have the time to replace quickly and felt it would be better to reintroduce them as part of a grander scheme instead.

Apologies about this. It's on our radar!

**Comment by lgvr123** [comment]
Shouldn't this one be labelled "regression_ms3" ?

**Comment by cbjeukendrup** [comment]
Issues with the "Feature requests" label are handled separately from the "backlog". Every release, we pick some issues from the backlog and some feature requests. 
The main reason that we had to move it out of the backlog is that we were hitting the maximum number of issues that can be in one GitHub Project (namely 1200).

**Comment by skybaks** [comment]
Just ran into this problem and was surprised to see this issue still out here. Is the hard part defining what the interface should look like or is that straightforward and its just a matter of time and competing priorities?

**Comment by MarcSabatella** [comment]
FYI, it *is* quite possible to mute individual voices, just not via the Mixer.  Instead, use the "View / Selection filter" to exclude the voices you wish to hear from the selection, then uncheck the "Play" property.  The existence of a very straightforward alternative method to achieve the result is likely also part of why this hasn't been prioriized higher thus far.

**Comment by johann-petrak** [comment]
> This was a MIDI-based capability we had to drop for 4.0 but will re-introduce in later versions.
> 
Sadly, the MIDI-based capabilities have not seen any improvements since the initial release of MS4 at all and are still not at par with MS version 3. 
A great use of MS3 for musicians was that multiple voice or multiple instrument scores could be used for practising/performing "minus-one" style on MIDI instruments which provides much much better sound quality that whatever audio MS can produce. This is still not possible with MS4.

**Comment by MarcSabatella** [comment]
See my comment above - it is certainly still possible to mute individual voices.  It just doesn’t happen to involve the Mixer.

**Comment by Jojo-Schmitz** [comment]
That method os just very likely to effect the score itself, not just it's **current** playback.
Also it doesn't allos for different volumers for different voices, just off/on
This is (one of) the main reason(s) why I still stick to 3.x

**Comment by skybaks** [comment]
Yeah, i see your workaround. Maybe my terminology is off here but my personal hangup is that you have to mutate the score itself to acheive the result of silencing one voice, vs the idea of leaving the score untouched and only changingthe output.
Both cases probably modify the save file but the semantic difference is slightly irksome to me.

**Comment by Tantacrul** [comment]
It's not really about whether or not we desired the feature - we did. It's a case similar to the situation with all scores being tabs in one instance of MuseScore. It was technically incompatible with what was needed to make MuseSounds work.

However, we've always wanted to do this. We just need to build the feature on top of our existing tech. I would kinda like it in the mixer myself tbh.

@bkunda - when the heat is a bit lower on MSS4.6, it'd be good to discuss this as a potential community release candidate. I'm not sure how it stacks up in priority compared to other desired things... but it'd be nice for sure. May be easy. May be quite rough.

**Comment by skybaks** [comment]
I do have some interest in attempting to add this myself if you could drop me any pointers or suggestions for the implementation

**Comment by Tantacrul** [comment]
THat's cool. I'll ask our resident audio experts who can let us know.

**Comment by MarcSabatella** [comment]
I would have to imagine the implementation would be trivial, once a design is decided upon.  From my read of traffic on forums and social media etc, it's up there with the image capture tool as one of the most sorely-missed features from MU3.

But to be clear: mute state *is* saved with the score even in MU3, and is also reflected in export, etc.  So the distinction between affecting the "play" property affecting score itself versus the "mute" button affecting current playback only is almost entirely artificial.  The one respect it isn't is that if you had some specific notes you wanted to *never* play for whatever reason, there wouldn't be a way to differentiate that from the ones you are *temporarily* disabling.  So if you tried to reverse the process to unmute the voice, you'd end up hearing those notes that had previously been disabled "permanently*.  Here, undo or reverting to a previously-saved version would be the solution.

Note that MU3 didn't allow for volume control on the voices either.  That would of course be a wonderful addition, though.  Meanwhile, the method of using the selection filter to isolate a voice and then overriding the *velocity* instead of *play* property would achieve the same result.  This works in both MU3 and MU4 - at least for soundfonts (which is all MU3 supports anyhow).  Not sure if the proposed automation lanes would provide a method of working by voice, but that would definitely be an even more welcome improvement!

So to me, adding a simple set of by-voice and by-staff (for grand staff instruments) mute buttons - maybe a popup on the main Mute button - would be incredibly useful for 4.6 if at all possible.  Then maybe 5.x could provide additional control via sliders or automation lanes.

**Comment by Jojo-Schmitz** [comment]
> Note that MU3 didn't allow for volume control on the voices either. 

It did, via using channels, e.g via the S/A staff text from thr palette and its channels for od voices (Soprano) and even voices (Alto) Same for T/B, T/L and B/B, to cover Closed Score SATB and Barbershop
